src/ray/core_worker/task_interface.cc

```diff
@@ -100,64 +99,67 @@ CoreWorkerTaskInterface::CoreWorkerTaskInterface(
                                new CoreWorkerRayletTaskSubmitter(raylet_client)));
 }
 
-Status CoreWorkerTaskInterface::SubmitTask(const RayFunction &function,
-                                           const std::vector<TaskArg> &args,
-                                           const TaskOptions &task_options,
-                                           std::vector<ObjectID> *return_ids) {
+rpc::TaskSpec CoreWorkerTaskInterface::CreateCommonTaskSpecMessage(
```

Comment:
This should move into the same module as `BuildCommonTaskSpec`.

Reply:
This method depends on some internal states of `CoreWorkerTaskInterface`, e.g., `worker_context_`. Its purpose is is to reduce duplicate code in `SubmitTask`, `SubmitActorTask`, and `CreateActor`.